samples/dfu/src/main.c

```diff
@@ -146,9 +146,40 @@ static int golioth_desired_update(struct golioth_req_rsp *rsp)
 	return 0;
 }
 
+static int async_error_handler(struct golioth_req_rsp *rsp)
+{
+	if (rsp->err) {
+		LOG_ERR("Async firmware report failed: %d", rsp->err);
+		return rsp->err;
+	}
+
+	LOG_INF("Async firmware report succeeded");
+	return 0;
+}
+
 static void golioth_on_connect(struct golioth_client *client)
 {
 	int err;
+	static bool first_contact = true;
```

Comment:
Please, no `static` variables within function body.

Reply:
Thanks for catching this one. I reimplemented this using a semaphore. Please take a look and let me know if this is an okay approach.